thrust/testing/functional.cu

```diff
@@ -180,21 +180,50 @@ DECLARE_UNARY_ARITHMETIC_FUNCTIONAL_UNITTEST(negate, Negate);
 _CCCL_DIAG_POP
 DECLARE_UNARY_LOGICAL_FUNCTIONAL_UNITTEST(logical_not, LogicalNot);
 
+// TODO(bgruber): replace by cuda::std::as_const in C++14
+template <class _Tp>
+typename ::cuda::std::add_const<_Tp>::type& as_const(_Tp& __t) noexcept
+{
+  return __t;
+}
+
 // Ad-hoc testing for other functionals
-template <class Vector>
 THRUST_DISABLE_BROKEN_GCC_VECTORIZER void TestIdentityFunctional()
 {
-  using T = typename Vector::value_type;
+  int i    = 42;
+  double d = 3.14;
```

Comment:
NVCC complains that d is not used, maybe add an `ASSERT_EQUAL` what you would expect?